client/jetpack-connect/signup.js

```diff
@@ -212,7 +260,111 @@ export class JetpackSignup extends Component {
 			</LoggedOutFormLinks>
 		);
 	}
+
+	renderWooDna() {
+		const { authQuery, isFullLoginFormVisible, translate, usernameOrEmail } = this.props;
+		const { isCreatingAccount, signUpUsernameOrEmail } = this.state;
+		let header, subHeader, content;
+		const footerLinks = [];
+		const email = signUpUsernameOrEmail || usernameOrEmail || authQuery.userEmail;
+		const wooDna = this.getWooDnaConfig();
+		let pageTitle;
+
+		if ( this.state.showWooDnaLoginForm ) {
+			if ( isFullLoginFormVisible ) {
```

Comment:
This is basically the difference between the "enter your email" screen and the "log in" screen right? Maybe we could either add a comment or change the variable name to something like `! isLoggedOutScreen`. Or we could designate three states in the component local state for the type of Woo DNA screen we're on - "log in", "sign up" or "logged out" (or "ask for email"). I commented about having a non-binary state variable this like this in another comment, too.

Reply:
> This is basically the difference between the "enter your email" screen and the "log in" screen right? 

Yes.

> change the variable name to something like `! isLoggedOutScreen`.

Both screens are "logged out" screens. Really, both screens are part of the "login" flow, just different steps on it. I understand that `isFullLoginFormVisible` is not the perfect variable name but it's the best I could come up with. If you can't think of anything better, I'll add a clarifying comment :)

> Or we could designate three states in the component local state for the type of Woo DNA screen we're on

I'm not a fan of this option. The `isFullLoginFormVisible` prop is derived from the global Redux state, so we shouldn't store its value in the local component state.